dotnet/src/Azure.Iot.Operations.Connector/Models/EndpointIdentity.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.Iot.Operations.Connector.Models;
+
+/// <summary>
+/// Represents endpoint information for CloudEvents correlation.
+/// Captures the protocol-specific connection details for a device endpoint.
+/// </summary>
+public class EndpointIdentity
+{
+    /// <summary>
+    /// Endpoint name (e.g., "default", "OpcUaEndpoint1").
+    /// This is a human-readable identifier for the endpoint configuration.
+    /// </summary>
+    public required string Name { get; init; }
+    
+    /// <summary>
+    /// Protocol-specific identifier used to uniquely identify the endpoint.
+    /// Examples:
+    /// - OPC UA: "opc.tcp://192.168.1.10:4840/UA/Server"
+    /// - REST/SSE: "https://api.example.com/v1/telemetry"
+    /// - Modbus: "modbus.tcp://192.168.1.20:502"
+    /// </summary>
+    public required string ProtocolIdentifier { get; init; }
```

Comment:
Isn't this just the [endpoint address](https://github.com/Azure/iot-operations-sdks/blob/e77761b4f075504bbe54805f048b017bec0fa933/dotnet/src/Azure.Iot.Operations.Services/AssetAndDeviceRegistry/Models/DeviceOutboundEndpoint.cs#L5)? If so, then, again, this data type is already modeled in the ADR service so we probably don't need to model it again in the connector package